### DISCUSSION TITLE
Possibility to wrap FluentTreeItems in a <div>?

### DISCUSSION THREAD

**Discussion by sahra-marie** [microsoft/fluentui-blazor#2489] [Q&A] [answered]
Hi there!
I'm looking for a possibility to wrap the children of a FluentTreeItem in a \<div\> without messing up the style.
I'm not entirely sure if this is a problem of the blazor component or the underlying web-components of fluentui but maybe someone else knows this?

I need the possibility to write my code like this:

<FluentTreeView>

        <FluentTreeItem Text="folder">
            <div>
                <FluentTreeItem Text="child1"></FluentTreeItem>
                <FluentTreeItem Text="child2"></FluentTreeItem>
                <FluentTreeItem Text="child3"></FluentTreeItem>
            </div>
        </FluentTreeItem>
</FluentTreeView>

But this is the outcome:
![image](https://github.com/user-attachments/assets/2483fd48-3737-4a64-b192-9ea1cb908ba0)
Instead of this:
![Screenshot 2024-08-07 150359](https://github.com/user-attachments/assets/1f20b9b7-b4b5-45e1-bdd7-8c9556d8fe68)

(Side note why I need this: I'm hoping that the \<div\> will make SortableJS work for the tree. In the end the \<div\> should get a class that allows me to find the elements via js, as this doesn't seem to work properly when setting the class directly on the FluentTreeItem.)

**Comment by dvoituron** [comment, marked as answer]
Unfortunately, I don't think there's anything we can do. This seems to be a limitation of the WebComponents currently in use.
The first example is well displayed (by `div`) but the second doesn't render correctly.

```html
  <fluent-tree-view>
    <fluent-tree-item>One</fluent-tree-item>
    <fluent-tree-item>Two
      <fluent-tree-item>A</fluent-tree-item>
      <fluent-tree-item>B</fluent-tree-item>
    </fluent-tree-item>
  </fluent-tree-view>
  
  <hr />
  
  <fluent-tree-view>
    <fluent-tree-item>One</fluent-tree-item>
    <fluent-tree-item>Two
    <div>    
        <fluent-tree-item>A</fluent-tree-item>
        <fluent-tree-item>B</fluent-tree-item>
      </div>
    </fluent-tree-item>
  </fluent-tree-view>
```

![image](https://github.com/user-attachments/assets/f0641bf7-1961-4297-a127-ba661da008e4)

See https://jsfiddle.net/dvoituron/skzbm6p5/106/
